src/make_zero.jl

```diff
@@ -232,11 +230,10 @@ end
         end
         return y
     end
-
     if nf == 0
-        return prev
+        # nothing to do, assume inactive
+        return copy_if_inactive ? Base.deepcopy_internal(prev, seen) : prev
```

Comment:
Do we need to do this though if there are no fields? This shouldn’t be necessary

Reply:
I was thinking that if you do it might be with some unorthodox type that holds potentially mutable data that's not exposed as a field, similar to `Array` which doesn't have fields but still has data. So to be safe you should respect `copy_if_inactive`. But I haven't ever been able to come up with an example that hits this branch, since empty structs go straight to `guaranteed_const_nongen`, so it's a rather hypothetical concern.

Reply:
So as long as we handle all primitive container types (which we do here already), it should be fine.

Reply:
Since all user defined types have to have fields

Reply:
To have this behavior**

Reply:
Added back the simpler variant. Really curious if it's possible to hit this branch at all.